src/main/java/games/strategy/engine/lobby/server/db/UserController.java

```diff
@@ -72,56 +74,64 @@ public boolean doesUserExist(final String userName) {
 
   @Override
   public void updateUser(final DBUser user, final HashedPassword hashedPassword) {
+    updateUserWithExactString(user, hashedPassword.value);
+  }
+
+  @Override
+  public void updateUser(final DBUser user, final String password) {
```

Comment:
These extra methods are what make me really request changes. The logic here needs to be made more agnostic to how the password is encrypted, it is not the responsibility of this layer. Push the bcrypt logic to the caller, keep the existing APIs please.